src/models/cooperation.js

```diff
@@ -114,14 +114,13 @@ const cooperationSchema = new Schema(
             _id: false,
             resource: {
               type: Schema.Types.Mixed,
-              required: [true, FIELD_CANNOT_BE_EMPTY('activity resource')]
+              values: [LESSONS, QUIZZES, ATTACHMENTS]
             },
             resourceType: {
               type: String,
-              required: [true, FIELD_MUST_BE_SELECTED('activity resource type')],
               enum: {
-                values: [LESSON, QUIZ, ATTACHMENT],
-                message: ENUM_CAN_BE_ONE_OF('activity resource type', [LESSON, QUIZ, ATTACHMENT])
+                values: [LESSONS, QUIZZES, ATTACHMENTS],
+                message: ENUM_CAN_BE_ONE_OF('activity resource type', [LESSONS, QUIZZES, ATTACHMENTS])
```

Comment:
I suggest creating separate array for resourses like following:

```jsx
const resources = [LESSONS, QUIZZES, ATTACHMENTS]
```

Reply:
fixed